scripts/rosversion

```diff
@@ -68,47 +69,75 @@ def get_distro_name_from_roscore():
         traceback.print_exc()
 
 
-if len(sys.argv) == 2:
-    stack_name = sys.argv[1]
+def get_version_printer(skipnewline=False):
+    """
+    @return: A method object to print the version string.
+    """
+    if skipnewline:
+        return sys.stderr.write
+    else:
+        return print
 
-    if stack_name == '-d' or stack_name == '--distro':
-        if 'ROS_DISTRO' in os.environ:
-            distro_name = os.environ['ROS_DISTRO']
-        else:
-            distro_name = get_distro_name_from_roscore()
-        if not distro_name:
-            print('<unknown>')
-        else:
-            print(distro_name)
-        sys.exit(0)
+parser = argparse.ArgumentParser(description=USAGE)
+if len(sys.argv) < 2:
+    parser.print_usage()
+    sys.exit(1)
+group = parser.add_mutually_exclusive_group()
+group.add_argument(
+    'package',
+    help="Package name. This name must be in ROS format, not OS package format. "
+         "E.g. CORRECT: roscpp, INCORRECT ros-kinetic-roscpp. See ROS naming "
+         "convention http://wiki.ros.org/ROS/Patterns/Conventions#Packages",
```

Comment:
I don't think this level of verbosity is necessary here. I boiled it down to mention `roscpp` as an example which should be clear enough.